test/FlowManyTest.cpp

```diff
@@ -104,122 +105,126 @@ SCENARIO("flow many immediate cancellation", "[flow][sender]") {
+//             if (requested < 1 || data.stop) {return;}
+//             // submit work to happen later
+//             auto datal = std::move(data);
+//             data.valid = false;
+//             tr |
+//                 op::submit_after(
+//                     100ms,
+//                     [data = std::move(datal)](auto) mutable {
+//                       // check boolean to select signal
+//                       if (!data.stop) {
+//                         ::mi::set_next(data.out, 42);
+//                       }
+//                       ::mi::set_done(data.out);
+//                     });
+//           },
+//           [&](auto& data, auto e) noexcept {
+//             signals += 100000;
+//             if(data.stop || !data.valid) {return;}
+//             data.stop = true;
+//             ::mi::set_done(data.out);
+//           },
+//           [&](auto& data) {
+//             signals += 10000;
+//             if(data.stop || !data.valid) {return;}
+//             data.stop = true;
+//             ::mi::set_done(data.out);
+//           });
+//
+//       tr |
+//           op::submit([up = std::move(up)](auto tr) mutable {
+//             // pass reference for cancellation.
+//             ::mi::set_starting(up.data().out, std::move(up));
+//           });
+//     });
+//
+//     WHEN("submit is applied and cancels the producer") {
+//       f |
+//           op::submit(
+//               mi::on_next([&](int) { signals += 100; }),
+//               mi::on_error([&](auto) noexcept { signals += 1000; }),
+//               mi::on_done([&]() { signals += 1; }),
+//               // stop producer before it is scheduled to run
+//               mi::on_starting([&](auto up) {
+//                 signals += 10;
+//                 tr | op::submit_after(50ms, [up = std::move(up)](auto) mutable {
+//                   ::mi::set_done(up);
+//                 });
+//               }));
+//
+//       THEN(
+//           "the starting, up.done and out.done signals are each recorded once") {
+//         REQUIRE(signals == 10011);
+//       }
+//     }
+//
+//     WHEN("submit is applied and cancels the producer late") {
+//       f |
+//           op::submit(
+//               mi::on_next([&](int) { signals += 100; }),
+//               mi::on_error([&](auto) noexcept { signals += 1000; }),
+//               mi::on_done([&]() { signals += 1; }),
+//               // do not stop producer before it is scheduled to run
+//               mi::on_starting([&](auto up) {
+//                 signals += 10;
+//                 mi::set_next(up, 1);
+//                 tr |
+//                     op::submit_after(250ms, [up = std::move(up)](auto) mutable {
+//                       ::mi::set_done(up);
+//                     });
+//               }));
+//
+//       THEN(
+//           "the starting, up.done and out.value signals are each recorded once") {
+//         REQUIRE(signals == 1010111);
+//       }
+//     }
+//   }
+// }
```

Comment:
Why is this test commented out?

Reply:
The trampoline tests are commented out because they depend on trampoline being a timesender. I cannot use trampoline with time_source, so I would have to revert to built in support for time in the trampoline to enable these